src/rgw/rgw_sal.h

```diff
@@ -1109,6 +1109,11 @@ class Object {
     /** Get a unique copy of this object */
     virtual std::unique_ptr<Object> clone() = 0;
 
+    /** get the trace stored in the object */
+    virtual jspan_context& get_trace() = 0;
+    /** set a trace to be stored in the object */
+    virtual void set_trace (jspan_context&& _trace_ctx) = 0;
+
```

Comment:
i would really prefer not to put this in `sal::Object`. please just pass the span into whichever operations require it. for `UpdateIndex`, that's probably just `sal::Writer::complete()`